workflow-ui/core-common/src/main/java/com/squareup/workflow1/visual/WithName.kt

```diff
@@ -0,0 +1,41 @@
+package com.squareup.workflow1.visual
+
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * Base type for [Wrapper]s that are used to change the
+ * [compatibility][com.squareup.workflow1.ui.compatible] of renderings
+ * that do not implement [Compatible][com.squareup.workflow1.ui.Compatible]
+ * themselves.
+ */
+@WorkflowUiExperimentalApi
+public abstract class WithName<W : Any>(
+  wrapped: W,
+  name: String
+) : Wrapper<W>(wrapped) {
+  public final override val name: String by lazy {
+    require(name.isNotBlank()) { "name must not be blank." }
+    name
+  }
+}
```

Comment:
nit: Maybe add a `fun T.withName(name): WithName<T>` convenience?

Reply:
I don't think that's possible, since `WithName` is an abstract class. We can't know what concrete subtype of `WithName` to instantiate.